src/autodiff/Gradient.cpp

```diff
@@ -112,19 +111,22 @@ Profiler& Gradient::GetProfiler() {
 }
 
 void Gradient::Compute() {
+  // Computes the gradient and assigns it to m_g.
```

Comment:
If this comment applies to the whole function, it should be a Doxygen comment in the header.